Build failure: i3ipc (Python)

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

build `i3ipc` python package

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/305912180

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gla6rf6cqrpw68b7sdl06i7453hh4w6b-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/test/test_window.py"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation python3.13-i3ipc-2.2.1: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'test/i3.config')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing setuptoolsBuildPhase
setup.py build flags: ''
/nix/store/zg70iz40mx8k33g553zf62wq7101c6k0-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
!!

        ********************************************************************************
        Please consider removing the following classifiers in favor of a SPDX license expression:

        License :: OSI Approved :: BSD License

        See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
        ********************************************************************************

!!
  self._finalize_license_expression()
running bdist_wheel
running build
running build_py
creating build/lib/i3ipc
copying i3ipc/model.py -> build/lib/i3ipc
copying i3ipc/events.py -> build/lib/i3ipc
copying i3ipc/__version__.py -> build/lib/i3ipc
copying i3ipc/connection.py -> build/lib/i3ipc
copying i3ipc/__init__.py -> build/lib/i3ipc
copying i3ipc/con.py -> build/lib/i3ipc
copying i3ipc/replies.py -> build/lib/i3ipc
creating build/lib/i3ipc/_private
copying i3ipc/_private/sync.py -> build/lib/i3ipc/_private
copying i3ipc/_private/pubsub.py -> build/lib/i3ipc/_private
copying i3ipc/_private/__init__.py -> build/lib/i3ipc/_private
copying i3ipc/_private/types.py -> build/lib/i3ipc/_private
creating build/lib/i3ipc/aio
copying i3ipc/aio/connection.py -> build/lib/i3ipc/aio
copying i3ipc/aio/__init__.py -> build/lib/i3ipc/aio
running egg_info
creating i3ipc.egg-info
writing i3ipc.egg-info/PKG-INFO
writing dependency_links to i3ipc.egg-info/dependency_links.txt
writing requirements to i3ipc.egg-info/requires.txt
writing top-level names to i3ipc.egg-info/top_level.txt
writing manifest file 'i3ipc.egg-info/SOURCES.txt'
reading manifest file 'i3ipc.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'i3ipc.egg-info/SOURCES.txt'
/nix/store/zg70iz40mx8k33g553zf62wq7101c6k0-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/cmd.py:90: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        By 2025-Oct-31, you need to update your project and remove deprecated calls
        or your builds will no longer be supported.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/i3ipc
copying build/lib/i3ipc/__version__.py -> build/bdist.linux-x86_64/wheel/./i3ipc
creating build/bdist.linux-x86_64/wheel/i3ipc/_private
copying build/lib/i3ipc/_private/__init__.py -> build/bdist.linux-x86_64/wheel/./i3ipc/_private
copying build/lib/i3ipc/_private/sync.py -> build/bdist.linux-x86_64/wheel/./i3ipc/_private
copying build/lib/i3ipc/_private/pubsub.py -> build/bdist.linux-x86_64/wheel/./i3ipc/_private
copying build/lib/i3ipc/_private/types.py -> build/bdist.linux-x86_64/wheel/./i3ipc/_private
creating build/bdist.linux-x86_64/wheel/i3ipc/aio
copying build/lib/i3ipc/aio/connection.py -> build/bdist.linux-x86_64/wheel/./i3ipc/aio
copying build/lib/i3ipc/aio/__init__.py -> build/bdist.linux-x86_64/wheel/./i3ipc/aio
copying build/lib/i3ipc/replies.py -> build/bdist.linux-x86_64/wheel/./i3ipc
copying build/lib/i3ipc/__init__.py -> build/bdist.linux-x86_64/wheel/./i3ipc
copying build/lib/i3ipc/model.py -> build/bdist.linux-x86_64/wheel/./i3ipc
copying build/lib/i3ipc/con.py -> build/bdist.linux-x86_64/wheel/./i3ipc
copying build/lib/i3ipc/connection.py -> build/bdist.linux-x86_64/wheel/./i3ipc
copying build/lib/i3ipc/events.py -> build/bdist.linux-x86_64/wheel/./i3ipc
running install_egg_info
Copying i3ipc.egg-info to build/bdist.linux-x86_64/wheel/./i3ipc-2.2.1-py3.13.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/i3ipc-2.2.1.dist-info/WHEEL
creating 'dist/i3ipc-2.2.1-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'i3ipc/__init__.py'
adding 'i3ipc/__version__.py'
adding 'i3ipc/con.py'
adding 'i3ipc/connection.py'
adding 'i3ipc/events.py'
adding 'i3ipc/model.py'
adding 'i3ipc/replies.py'
adding 'i3ipc/_private/__init__.py'
adding 'i3ipc/_private/pubsub.py'
adding 'i3ipc/_private/sync.py'
adding 'i3ipc/_private/types.py'
adding 'i3ipc/aio/__init__.py'
adding 'i3ipc/aio/connection.py'
adding 'i3ipc-2.2.1.dist-info/licenses/LICENSE'
adding 'i3ipc-2.2.1.dist-info/METADATA'
adding 'i3ipc-2.2.1.dist-info/WHEEL'
adding 'i3ipc-2.2.1.dist-info/top_level.txt'
adding 'i3ipc-2.2.1.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed i3ipc-2.2.1-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/z7c432x27k4fxg0dqhizlsrfbv78j1za-python3.13-i3ipc-2.2.1
checking for references to /build/ in /nix/store/z7c432x27k4fxg0dqhizlsrfbv78j1za-python3.13-i3ipc-2.2.1...
patching script interpreter paths in /nix/store/z7c432x27k4fxg0dqhizlsrfbv78j1za-python3.13-i3ipc-2.2.1
stripping (with command strip and flags -S -p) in  /nix/store/z7c432x27k4fxg0dqhizlsrfbv78j1za-python3.13-i3ipc-2.2.1/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/3w98mcy9bpk0my9p8zba4qxpflx8c8br-python3.13-i3ipc-2.2.1-dist
checking for references to /build/ in /nix/store/3w98mcy9bpk0my9p8zba4qxpflx8c8br-python3.13-i3ipc-2.2.1-dist...
patching script interpreter paths in /nix/store/3w98mcy9bpk0my9p8zba4qxpflx8c8br-python3.13-i3ipc-2.2.1-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
[1m============================= test session starts ==============================[0m
platform linux -- Python 3.13.6, pytest-8.4.1, pluggy-1.6.0
rootdir: /build/source
configfile: pytest.ini
plugins: xvfb-3.1.1
[1mcollecting ... [0m[1m
collecting 0 items                                                             [0m[1m
collecting 10 items                                                            [0m[1m
collected 24 items                                                             [0m

test/aio/test_event_exceptions.py [31mF[0m[31m                                      [  4%][0m
test/aio/test_get_bindings_modes.py [31mF[0m[31m                                    [  8%][0m
test/aio/test_get_config.py [31mF[0m[31m                                            [ 12%][0m
test/aio/test_get_marks.py [31mF[0m[31m                                             [ 16%][0m
test/aio/test_leaves.py [31mF[0m[31m                                                [ 20%][0m
test/aio/test_output.py [31mF[0m[31m                                                [ 25%][0m
test/aio/test_requests.py [31mF[0m[31m                                              [ 29%][0m
test/aio/test_restart.py [31mF[0m[31m                                               [ 33%][0m
test/aio/test_scratchpad.py [31mF[0m[31m                                            [ 37%][0m
test/aio/test_ticks.py [31mF[0m[31m                                                 [ 41%][0m
test/aio/test_window.py [31mF[0m[31mF[0m[31mF[0m[31m                                              [ 54%][0m
test/aio/test_workspace.py [31mF[0m[31m                                             [ 58%][0m
test/test_event_exceptions.py [32m.[0m[31m                                          [ 62%][0m
test/test_get_bindings_modes.py [32m.[0m[31m                                        [ 66%][0m
test/test_get_config.py [32m.[0m[31m                                                [ 70%][0m
test/test_get_marks.py [32m.[0m[31m                                                 [ 75%][0m
test/test_leaves.py [32m.[0m[31m                                                    [ 79%][0m
test/test_restart.py [32m.[0m[31m                                                   [ 83%][0m
test/test_ticks.py [32m.[0m[31m                                                     [ 87%][0m
test/test_window.py [32m.[0m[32m.[0m[32m.[0m[31m                                                  [100%][0m

=================================== FAILURES ===================================
[31m[1m__________________ TestEventExceptions.test_event_exceptions ___________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_____________________ TestBindingModes.test_binding_modes ______________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m________________________ TestGetConfig.test_get_config _________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_________________________ TestGetMarks.test_get_marks __________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_______________________ TestLeaves.test_workspace_leaves _______________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m____________________________ TestOutput.test_output ____________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_________________________ TestResquests.test_requests __________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_______________________ TestRestart.test_auto_reconnect ________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m________________________ TestScratchpad.test_scratchpad ________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m__________________________ TestTicks.test_tick_event ___________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_________________________ TestWindow.test_window_event _________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m____________________ TestWindow.test_detailed_window_event _____________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m____________________________ TestWindow.test_marks _____________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[31m[1m_________________________ TestWorkspace.test_workspace _________________________[0m
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
[33m=============================== warnings summary ===============================[0m
../../nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: timeout
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

test/aio/test_event_exceptions.py:15
  /build/source/test/aio/test_event_exceptions.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_get_bindings_modes.py:7
  /build/source/test/aio/test_get_bindings_modes.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_get_config.py:9
  /build/source/test/aio/test_get_config.py:9: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_get_marks.py:7
  /build/source/test/aio/test_get_marks.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_leaves.py:7
  /build/source/test/aio/test_leaves.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_output.py:7
  /build/source/test/aio/test_output.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_requests.py:11
  /build/source/test/aio/test_requests.py:11: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_restart.py:7
  /build/source/test/aio/test_restart.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_scratchpad.py:7
  /build/source/test/aio/test_scratchpad.py:7: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_ticks.py:15
  /build/source/test/aio/test_ticks.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_window.py:9
  /build/source/test/aio/test_window.py:9: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_window.py:29
  /build/source/test/aio/test_window.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_window.py:68
  /build/source/test/aio/test_window.py:68: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_workspace.py:17
  /build/source/test/aio/test_workspace.py:17: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

test/aio/test_event_exceptions.py::TestEventExceptions::test_event_exceptions
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestEventExceptions' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_get_bindings_modes.py::TestBindingModes::test_binding_modes
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestBindingModes' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_get_config.py::TestGetConfig::test_get_config
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestGetConfig' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_get_marks.py::TestGetMarks::test_get_marks
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestGetMarks' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_leaves.py::TestLeaves::test_workspace_leaves
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestLeaves' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_output.py::TestOutput::test_output
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestOutput' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_requests.py::TestResquests::test_requests
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestResquests' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_restart.py::TestRestart::test_auto_reconnect
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestRestart' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_scratchpad.py::TestScratchpad::test_scratchpad
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestScratchpad' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_ticks.py::TestTicks::test_tick_event
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestTicks' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_window.py::TestWindow::test_window_event
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestWindow' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

test/aio/test_workspace.py::TestWorkspace::test_workspace
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestWorkspace' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
  See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[36m[1m=========================== short test summary info ============================[0m
[31mFAILED[0m test/aio/test_event_exceptions.py::[1mTestEventExceptions::test_event_exceptions[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_get_bindings_modes.py::[1mTestBindingModes::test_binding_modes[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_get_config.py::[1mTestGetConfig::test_get_config[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_get_marks.py::[1mTestGetMarks::test_get_marks[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_leaves.py::[1mTestLeaves::test_workspace_leaves[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_output.py::[1mTestOutput::test_output[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_requests.py::[1mTestResquests::test_requests[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_restart.py::[1mTestRestart::test_auto_reconnect[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_scratchpad.py::[1mTestScratchpad::test_scratchpad[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_ticks.py::[1mTestTicks::test_tick_event[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_window.py::[1mTestWindow::test_window_event[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_window.py::[1mTestWindow::test_detailed_window_event[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_window.py::[1mTestWindow::test_marks[0m - Failed: async def functions are not natively supported.
[31mFAILED[0m test/aio/test_workspace.py::[1mTestWorkspace::test_workspace[0m - Failed: async def functions are not natively supported.
[31m================== [31m[1m14 failed[0m, [32m10 passed[0m, [33m27 warnings[0m[31m in 5.97s[0m[31m ==================[0m
```

### Additional context

_No response_

### System metadata

```sh-session
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.42, NixOS, 25.11 (Xantusia), 25.11.20250819.2007595`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/pxnx7dbr0pz3qpgw2r315wdbb00afdw7-source`
```

### Notify maintainers

@vanzef 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSU@vanzef 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project. I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

hi, I think I get the same issue here, I have tried both workarounds, but I keep getting the same result

```sh
error: builder for '/nix/store/9pzslak7nadw30afwasddd36yn3wn3mi-python3.13-i3ipc-2.2.1.drv' failed with exit code 1;
       last 25 log lines:
       >   See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
       >     warnings.warn(
       >
       > test/aio/test_workspace.py::TestWorkspace::test_workspace
       >   /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/fixtures.py:1181: PytestRemovedIn9Warning: 'TestWorkspace' requested an async fixture 'i3', with no plugin or hook that handled it. This is usually an error, as pytest does not natively support it. This will turn into an error in pytest 9.
       >   See: https://docs.pytest.org/en/stable/deprecations.html#sync-test-depending-on-async-fixture
       >     warnings.warn(
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED test/aio/test_event_exceptions.py::TestEventExceptions::test_event_exceptions - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_get_bindings_modes.py::TestBindingModes::test_binding_modes - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_get_config.py::TestGetConfig::test_get_config - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_get_marks.py::TestGetMarks::test_get_marks - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_leaves.py::TestLeaves::test_workspace_leaves - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_output.py::TestOutput::test_output - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_requests.py::TestResquests::test_requests - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_restart.py::TestRestart::test_auto_reconnect - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_scratchpad.py::TestScratchpad::test_scratchpad - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_ticks.py::TestTicks::test_tick_event - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_window.py::TestWindow::test_window_event - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_window.py::TestWindow::test_detailed_window_event - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_window.py::TestWindow::test_marks - Failed: async def functions are not natively supported.
       > FAILED test/aio/test_workspace.py::TestWorkspace::test_workspace - Failed: async def functions are not natively supported.
       > ================== 14 failed, 10 passed, 27 warnings in 1.24s ==================
       For full logs, run:
         nix log /nix/store/9pzslak7nadw30afwasddd36yn3wn3mi-python3.13-i3ipc-2.2.1.drv
error: 1 dependencies of derivation '/nix/store/zk0sqnwi5736csgipjx4baj53wivw8yg-python3.13-powerline-2.8.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xdirn2jmqav3qvxy8q5dypgblnqbpnl6-X11-fonts.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6br7c79in65pmrm7ymi41b66wrnckpdg-fc-00-nixos-cache.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9lgvjvwb8k4vjgsyaixvhxyvw46xgc5b-steam-1.0.0.83-fhsenv-rootfs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qb2fnxk1ad6z51xxmnhy3xc2kf9nr7f1-steam-run-fhsenv-rootfs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ll86724qi6v26jgjiadp7wxmg6ngg8b8-xserver.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0aphlp2r71dz82ds59s9703c2kz95mjj-fontconfig-conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4kds7lmi283h7fsd8j93s2xxcmhdyb82-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6yrf6chc95pj3ygjr50qxzlbd6lpvfql-xserver-wrapper.drv' failed to build
error: 1 dependencies of derivation '/nix/store/r4nkbb4hxm0sbk30w65n17nzimhn23gl-nixos-system-nixosaku-25.11.20250827.8a6d542.drv' failed to build
Command 'nix --extra-experimental-features 'nix-command flakes' build --print-out-paths '/home/akunito/.dotfiles#nixosConfigurations."system".config.system.build.toplevel' --show-trace --impure --no-link' returned non-zero exit status 1.
```

any idea?

--- a/pkgs/development/python-modules/i3ipc/default.nix
+++ b/pkgs/development/python-modules/i3ipc/default.nix
@@ -4,6 +4,7 @@
   fetchFromGitHub,
   xorg,
   pytest,
+  pytest-asyncio,
   pytest-xvfb,
   i3,
   xlib,
@@ -23,12 +24,21 @@ buildPythonPackage rec {
     rev = "v${version}";
     sha256 = "13bzs9dcv27czpnnbgz7a037lm8h991c8gk0qzzk5mq5yak24715";
   };
+
+  patches = [
+    # Upstream expects a very old version of pytest-asyncio. This patch correctly
+    # decorates async fixtures using pytest-asyncio and configures `loop_scope`
+    # where needed.
+    ./fix-async-tests.patch
+  ];
+
   propagatedBuildInputs = [ xlib ];
 
   fontsConf = makeFontsConf { fontDirectories = [ ]; };
   FONTCONFIG_FILE = fontsConf; # Fontconfig error: Cannot load default config file
   nativeCheckInputs = [
     pytest
+    pytest-asyncio
     xdpyinfo
     pytest-xvfb
     xorg.xvfb

--- a/pkgs/development/python-modules/i3ipc/default.nix
+++ b/pkgs/development/python-modules/i3ipc/default.nix
@@ -2,28 +2,28 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  xorg,
-  pytest,
+  coreutils,
+  setuptools,
+  xlib,
+  fontconfig,
+  pytestCheckHook,
   pytest-asyncio,
   pytest-timeout,
   pytest-xvfb,
   i3,
-  xlib,
-  xdpyinfo,
-  makeFontsConf,
-  coreutils,
+  xorg,
 }:
 
 buildPythonPackage rec {
   pname = "i3ipc";
   version = "2.2.1";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchFromGitHub {
-    owner = "acrisci";
+    owner = "altdesktop";
     repo = "i3ipc-python";
-    rev = "v${version}";
-    sha256 = "13bzs9dcv27czpnnbgz7a037lm8h991c8gk0qzzk5mq5yak24715";
+    tag = "v${version}";
+    hash = "sha256-JRwipvIF1zL/x2A+xEJKEFV6BlDnv2Xt/eyIzVrSf40=";
   };
 
   patches = [
@@ -33,34 +33,40 @@ buildPythonPackage rec {
     ./fix-async-tests.patch
   ];
 
-  propagatedBuildInputs = [ xlib ];
-
-  fontsConf = makeFontsConf { fontDirectories = [ ]; };
-  FONTCONFIG_FILE = fontsConf; # Fontconfig error: Cannot load default config file
-  nativeCheckInputs = [
-    pytest
-    pytest-asyncio
-    xdpyinfo
-    pytest-timeout
-    pytest-xvfb
-    xorg.xvfb
-    i3
-  ];
-
   postPatch = ''
     substituteInPlace test/i3.config \
-      --replace /bin/true ${coreutils}/bin/true
+      --replace-fail /bin/true ${coreutils}/bin/true
   '';
 
-  checkPhase = ''
-    py.test --ignore=test/aio/test_shutdown_event.py \
-            --ignore=test/test_shutdown_event.py
-  '';
+  build-system = [ setuptools ];
+  dependencies = [ xlib ];
 
-  meta = with lib; {
+  # Fontconfig error: Cannot load default config file
+  env.FONTCONFIG_FILE = "${fontconfig.out}/etc/fonts/fonts.conf";
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    pytest-timeout
+    pytest-xvfb
+    i3
+    xorg.xdpyinfo
+    xorg.xvfb
+  ];
+
+  disabledTestPaths = [
+    # Timeout
+    "test/test_shutdown_event.py::TestShutdownEvent::test_shutdown_event_reconnect"
+    "test/aio/test_shutdown_event.py::TestShutdownEvent::test_shutdown_event_reconnect"
+  ];
+
+  pythonImportsCheck = [ "i3ipc" ];
+
+  meta = {
     description = "Improved Python library to control i3wm and sway";
-    homepage = "https://github.com/acrisci/i3ipc-python";
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ vanzef ];
+    homepage = "https://github.com/altdesktop/i3ipc-python";
+    changelog = "https://github.com/altdesktop/i3ipc-python/releases/tag/${src.tag}";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ vanzef ];
   };
 }

--- a/pkgs/development/python-modules/i3ipc/default.nix
+++ b/pkgs/development/python-modules/i3ipc/default.nix
@@ -7,6 +7,7 @@
   xlib,
   fontconfig,
   pytestCheckHook,
+  writableTmpDirAsHomeHook,
   pytest-asyncio,
   pytest-timeout,
   pytest-xvfb,
@@ -46,6 +47,7 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     pytestCheckHook
+    writableTmpDirAsHomeHook
     pytest-asyncio
     pytest-timeout
     pytest-xvfb
@@ -58,6 +60,9 @@ buildPythonPackage rec {
     # Timeout
     "test/test_shutdown_event.py::TestShutdownEvent::test_shutdown_event_reconnect"
     "test/aio/test_shutdown_event.py::TestShutdownEvent::test_shutdown_event_reconnect"
+    # Flaky
+    "test/test_window.py::TestWindow::test_detailed_window_event"
+    "test/aio/test_workspace.py::TestWorkspace::test_workspace"
   ];
 
   pythonImportsCheck = [ "i3ipc" ];

--- a/pkgs/development/python-modules/i3ipc/default.nix
+++ b/pkgs/development/python-modules/i3ipc/default.nix
@@ -5,6 +5,7 @@
   xorg,
   pytest,
   pytest-asyncio,
+  pytest-timeout,
   pytest-xvfb,
   i3,
   xlib,
@@ -40,6 +41,7 @@ buildPythonPackage rec {
     pytest
     pytest-asyncio
     xdpyinfo
+    pytest-timeout
     pytest-xvfb
     xorg.xvfb
     i3

--- a/pkgs/development/python-modules/i3ipc/fix-async-tests.patch
+++ b/pkgs/development/python-modules/i3ipc/fix-async-tests.patch
@@ -1,0 +1,28 @@
+diff --git a/test/aio/ipctest.py b/test/aio/ipctest.py
+index 88e4cda..3d0fd9c 100644
+--- a/test/aio/ipctest.py
++++ b/test/aio/ipctest.py
+@@ -1,5 +1,6 @@
+ from subprocess import Popen
+ import pytest
++import pytest_asyncio
+ 
+ from i3ipc.aio import Connection
+ from i3ipc import CommandReply
+@@ -19,7 +20,7 @@ class IpcTest:
+     def event_loop(self):
+         return asyncio.get_event_loop()
+ 
+-    @pytest.fixture(scope='class')
++    @pytest_asyncio.fixture(scope='class', loop_scope='class')
+     async def i3(self):
+         process = Popen(['i3', '-c', 'test/i3.config'])
+         # wait for i3 to start up
+diff --git a/pytest.ini b/pytest.ini
+index 1ea6b80..788bdac 100644
+--- a/pytest.ini
++++ b/pytest.ini
+@@ -1,2 +1,3 @@
+ [pytest]
+ timeout = 5
++asyncio_default_test_loop_scope = class

